R/categories.R

```diff
@@ -289,9 +291,11 @@ changeCategoryID <- function (variable, from, to) {
 
     ## Add new category
     newcat <- categories(variable)[[pos.from]]
+    if (newcat$id == newcat$numeric_value %||% FALSE) {
+        newcat$numeric_value <- to
```

Comment:
I wonder if we should have a warning here. I think this is probably the best solution but it's still somewhat counterintuitive behaviour so it might be good to have a warning. It probably is also a good idea to put a comment explaining why this change is here.